ocrd_models/ocrd_models/ocrd_file.py

```diff
@@ -13,40 +14,35 @@ class OcrdFile():
     Represents a single ``mets:file/mets:FLocat`` (METS file entry).
     """
 
-    #  @staticmethod
-    #  def create(mimetype, ID, url, local_filename):
-    #      el_fileGrp.SubElement('file')
-
-    def __init__(self, el, mimetype=None, pageId=None, loctype='OTHER', local_filename=None, mets=None, url=None, ID=None):
+    def __init__(self, el, mimetype=None, pageId=None, local_filename=None, mets=None, url=None, ID=None, loctype=None):
         """
         Args:
             el (LxmlElement): etree Element of the ``mets:file`` this represents. Create new if not provided
         Keyword Args:
             mets (OcrdMets): Containing :py:class:`ocrd_models.ocrd_mets.OcrdMets`.
             mimetype (string): ``@MIMETYPE`` of this ``mets:file``
             pageId (string): ``@ID`` of the physical ``mets:structMap`` entry corresponding to this ``mets:file``
-            loctype (string): ``@LOCTYPE`` of this ``mets:file``
             local_filename (string): Local filename
-            url (string): ``@xlink:href`` of this ``mets:file``
+            url (string): original ``@xlink:href`` of this ``mets:file``
+            local_filename (Path): ``@xlink:href`` pointing to the locally cached version of the file in the workspace
             ID (string): ``@ID`` of this ``mets:file``
+            loctype (string): DEPRECATED do not use
         """
         if el is None:
             raise ValueError("Must provide mets:file element this OcrdFile represents")
+        if loctype:
+            deprecation_warning("'loctype' is not supported in OcrdFile anymore, use 'url' or 'local_filename'")
         self._el = el
         self.mets = mets
         self.ID = ID
         self.mimetype = mimetype
-        self.local_filename = local_filename
-        self.loctype = loctype
         self.pageId = pageId
 
+        if local_filename:
+            self.local_filename = Path(local_filename)
```

Comment:
Should have been `str(Path(local_filename))`, right? @kba

Reply:
No, that was a conscious decision, though not optimally communicated. Changing the type helped me spot areas where `url` and `local_filename` were used interchangeably but I think it is a correct implementation that `local_filename` is a Path.

Reply:
Reverted in #1182